### DISCUSSION TITLE
Subclass methods

### DISCUSSION THREAD

**Discussion by kuraga** [coady/multimethod#112] [Q&A] [answered]
Good day! Thanks for your work!

Should I decorate the subclass' method?
How do I do it right?

```python
import multimethod

class A:
    @multimethod.multimethod
    def meth(self, arg: int):
        pass

    @multimethod.multimethod
    def meth(self, arg: str):
        pass

class B(A):
    ...
```

Thanks!

**Comment by coady** [comment, marked as answer]
```python
class B(A):
    @A.meth.register
    def _(self, arg: float): ...
```

You may also be interested in discussion #106, which is a similar issue.

**Reply by kuraga** [reply]
@coady , thanks!

But https://github.com/coady/multimethod/discussions/106#discussioncomment-7967839 suggests:

```python
class B(A):
    f = multimethod(A.f)

    @multimethod
    def f(self, x: int):
        return x
```

Which is correct?

**Reply by coady** [reply]
Both. The example in #106 assumes that the author can't change `A` to have multimethods.

**Reply by kuraga** [reply]
@coady , thanks!

P.S. Didn't see class examples in the docs. 'd be nice to have them...

**Reply by kuraga** [reply]
> P.S. Didn't see class examples in the docs. 'd be nice to have them...

> https://github.com/coady/multimethod/commit/a6cc05bfc62fe28ac3327e55c065eb16a1185490

@coady , thanks very much!
